### PR TITLE
mkdocs-include-markdown-plugin: init at 6.0.4

### PR DESCRIPTION
**Pull request by caarlos0**
## Description of changes

Adds yet another mkdocs plugin.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/mkdocs-include-markdown-plugin/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+buildPythonPackage rec {
+  pname = "mkdocs-include-markdown-plugin";
+  version = "6.0.4";
+
+  src = fetchFromGitHub {
+    owner = "mondeja";
+    repo = "mkdocs-include-markdown-plugin";
+    rev = "v${version}";
+    sha256 = "sha256-wHaDvF+QsEa3G5+q1ZUQQpVmwy+oRsSEq2qeJIJjFeY=";
```
Comment (WolfangAukang) [inline pkgs/development/python-modules/mkdocs-include-markdown-plugin/default.nix:17]:
```suggestion
    hash = "sha256-wHaDvF+QsEa3G5+q1ZUQQpVmwy+oRsSEq2qeJIJjFeY=";
```

File: pkgs/development/python-modules/mkdocs-include-markdown-plugin/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  };
+
+  pyproject = true;
+  doCheck = false;
```
Comment (WolfangAukang) [inline pkgs/development/python-modules/mkdocs-include-markdown-plugin/default.nix:21]: Specify on a comment above why the checks are disabled. The repo contains tests.

File: pkgs/development/python-modules/mkdocs-include-markdown-plugin/default.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+  nativeBuildInputs = [ hatchling mkdocs ];
+  propagatedBuildInputs = [ wcmatch ];
+
+  meta = with lib; {
+    description = "Mkdocs Markdown includer plugin.";
+    homepage = "https://github.com/mondeja/mkdocs-include-markdown-plugin";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ caarlos0 ];
```
Comment (WolfangAukang) [inline pkgs/development/python-modules/mkdocs-include-markdown-plugin/default.nix:30]:
```suggestion
    maintainers = with maintainers; [ caarlos0 ];
    changelog = "https://github.com/mondeja/mkdocs-include-markdown-plugin/releases/tag/${src.rev}";
```

File: pkgs/development/python-modules/mkdocs-include-markdown-plugin/default.nix [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  };
+
+  pyproject = true;
+  doCheck = false;
+
+  nativeBuildInputs = [ hatchling mkdocs ];
+  propagatedBuildInputs = [ wcmatch ];
+
+  meta = with lib; {
+    description = "Mkdocs Markdown includer plugin.";
```
Comment (WolfangAukang) [inline pkgs/development/python-modules/mkdocs-include-markdown-plugin/default.nix:27]:
```suggestion
    description = "Mkdocs Markdown includer plugin";
```

**Review by WolfangAukang — CHANGES_REQUESTED** [review]

**Comment by WolfangAukang** [comment]
Result of `nixpkgs-review pr 277351` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.mkdocs-include-markdown-plugin</li>
    <li>python310Packages.mkdocs-include-markdown-plugin.dist</li>
    <li>python311Packages.mkdocs-include-markdown-plugin</li>
    <li>python311Packages.mkdocs-include-markdown-plugin.dist</li>
  </ul>
</details>

**Comment by caarlos0** [comment]
@WolfangAukang updated

**Comment by caarlos0** [comment]
gently ping :)

**Comment by caarlos0** [comment]
gently ping @WolfangAukang

**Pull request closed without merging**